Fixed drawing of UI to make rows selectable when Process is hidden

MichaelZp0 opened the pull request:
Should fix #29 .

Also skips hidden columns in the copy rows function. I added this because I sometimes have to compare traces with a text editor and some columns make that pretty hard and they need to be filtered (aka Process/Thread IDs, Timestamps...).

I also renamed `i` to `rowIdx` as it is much easier to read that way.

Commit:
Fixed drawing of UI to make rows selectable when Process is hidden

brycehutchings commented:
I was testing this out but there is something weird going on. When I hide a column, the text in the next column goes empty. It makes me think the `ImGui.TableGetColumnFlags() & ImGuiTableColumnFlags.IsVisible` result is actually for an adjacent column?

brycehutchings commented:
> I was testing this out but there is something weird going on. When I hide a column, the text in the next column goes empty. It makes me think the `ImGui.TableGetColumnFlags() & ImGuiTableColumnFlags.IsVisible` result is actually for an adjacent column?

Ok actually that's not it. There can be are two TableNextColumn calls for a hidden column when the hidden column isn't the first one which is handled specially

brycehutchings commented:
I tried to get it this fix to work fully but I can't. It's a complicated area of the software.

brycehutchings reviewed (changes requested):
Hidden columns (besides the first) now cause the next column to be empty.

Commit:
Fixed hiding of values if column in middle is hidden

Commit:
Fixed message column having wrong context menu

MichaelZp0 commented:
Sorry, I only tested with hiding the first column and then everything worked just fine.

Yeah it was the TableNextColumn() calls the messed with the state, but I think I fixed it now.

There was also a second bug I found, that the right click menu of the last column header was showing the wrong content because it was thinking it was a data row and not a head row. I also fixed that by moving around the isRowHovered/hoveredCol code, because that relies on the full row selectable and would mess up if it was checked before the selectable was created.

brycehutchings reviewed (approved):
Thank you, this looks good!
